fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/AgilityContextTest.java

```diff
@@ -419,6 +421,80 @@ void testAgilityContextAtomicAttribute(boolean useAgile) {
+            keyAborted = subspace.pack(Tuple.from(2023, 3));
+            agilityContext.set(keyAborted, packedValue);
+            agilityContext.abortAndClose();
+            IntStream range = IntStream.rangeClosed(1, 10);
+            range.parallel().forEach(i -> {
+                if (i == 7) {
+                    agilityContext.applyInRecoveryPath(aContext -> {
+                        Assertions.assertNotNull(aContext);
+                        refInt.addAndGet(100);
+                        return CompletableFuture.completedFuture(null);
+                    }).join();
+                } else {
+                    Assertions.assertThrows(RecordCoreStorageException.class,
+                            () -> agilityContext.apply(aContext -> {
+                                refInt.set(i);
+                                return CompletableFuture.completedFuture(null);
+                            }).join());
+                }
+            });
+            assertEquals(100, refInt.get());
+            context.commit();
+        }
+    }
+
+    @Test
+    void testAgilityContextRecoveryPath2() {
+        final byte[] keyAborted;
+        final byte[] keySucceeds;
+        final byte[] keyFails;
+        final Tuple value = Tuple.from(800, "Green eggs and ham", 0);
+        byte[] packedValue = value.pack();
+        try (FDBRecordContext context = openContext()) {
+            final AgilityContext agilityContext = getAgilityContext(context, true);
+            final Subspace subspace = path.toSubspace(context);
+            keyAborted = subspace.pack(Tuple.from(2023, 3));
+            agilityContext.set(keyAborted, packedValue);
+            agilityContext.abortAndClose();
+            IntStream range = IntStream.rangeClosed(1, 10);
+            keySucceeds = this.path.toSubspace(context).pack(Tuple.from(prefix, "yes").pack());
+            keyFails = this.path.toSubspace(context).pack(Tuple.from(prefix, "no").pack());
+            range.parallel().forEach(i -> {
+                if (i == 7) {
+                    agilityContext.applyInRecoveryPath(aContext -> {
+                        Assertions.assertNotNull(aContext);
+                        context.ensureActive().set(keySucceeds, packedValue);
+                        return CompletableFuture.completedFuture(null);
+                    }).join();
+                } else {
+                    Assertions.assertThrows(RecordCoreStorageException.class,
+                            () -> agilityContext.apply(aContext -> {
+                                context.ensureActive().set(keyFails, packedValue);
+                                return CompletableFuture.completedFuture(null);
+                            }).join());
+                    Assertions.assertThrows(RecordCoreStorageException.class,
+                            () -> agilityContext.set(keyFails, packedValue));
+                }
+            });
+            context.commit();
```

Comment:
It seems strange to commit the caller context, is that what you want?

Reply:
Yes. I wanted to emulate a user that eventually commits the context he opened. As a read-only, it should be a no-op. Is it?